tests/foreman/maintain/test_service.py

```diff
@@ -0,0 +1,267 @@
+
+    :expectedresults: Satellite services should stop and start.
+    """
+    result = sat_maintain.cli.Service.stop()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    result = sat_maintain.cli.Service.start()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_service_stop_restart(sat_maintain):
+    """Disable services using satellite-maintain service
+
+    :id: dc60e388-f012-4164-a496-b12d6230cdc2
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service stop.
+        2. Run satellite-maintain service status --brief
+        3. Run satellite-maintain service restart
+        4. Run satellite-maintain service status --brief
+
+    :expectedresults: service should restart.
+
+    :BZ: 1626651
+    """
+    result = sat_maintain.cli.Service.stop()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    result = sat_maintain.cli.Service.restart()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_service_enable_disable(sat_maintain):
+    """Enable/Disable services using satellite-maintain service subcommand
+
+    :id: a0e0a052-0e21-465c-bb28-2e7613dbece6
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service disable
+        2. Run satellite-maintain service enable
+
+    :expectedresults: Services should enable/disable
+    """
+    if not is_open('BZ:1995783'):
+        result = sat_maintain.cli.Service.disable()
+        assert 'FAIL' not in result.stdout
+        assert result.status == 0
+    result = sat_maintain.cli.Service.enable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_foreman_service(request, sat_maintain):
+    """Validate httpd service should work as expected even stopping of the foreman service
+
+    :id: 08a29ea2-2e49-11eb-a22b-d46d6dd3b5b2
+
+    :steps:
+        1. Stop foreman service
+        2. Check the status for httpd should not have affect
+        3. Run satellite-maintain health check
+
+    :expectedresults: foreman service should restart correctly
+    """
+    result = sat_maintain.cli.Service.stop(options={'only': 'foreman'})
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'foreman' in result.stdout
```

Comment:
I think its good to check the actual status of foreman service through `Service.status` stdout than just chaking FAIL or inclusion of `foreman`.

Same for all the tests in this PR.

Reply:
Here Service.status also returns same thing as Service.stop, i don't think it will make any sense here

Reply:
```suggestion
    assert 'fail' not in result.stdout.lower()
    assert '<something >foreman <something>' in result.stdout 
```
Be specific with assertions and care about casing in assertions .